src/compiler/types.ts

```diff
@@ -2624,6 +2624,14 @@ namespace ts {
         /* @internal */ pragmas: PragmaMap;
         /* @internal */ localJsxNamespace?: __String;
         /* @internal */ localJsxFactory?: EntityName;
+
+        /*@internal*/ getExportedModulesFromDeclarationEmit?(): ExportedModulesFromDeclarationEmit;
+    }
+
+    /*@internal*/
+    export interface ExportedModulesFromDeclarationEmit {
+        exportedModuleSpecifiers: ReadonlyArray<StringLiteralLike>;

```

Comment:
From what I can tell, both of these just get mapped to a path eventually, via lookup (string ->) symbol -> sourcefile -> path. Rather than storing both symbols and string literals here, I'd rather we just stored the paths for all kinds directly. Minimally, just one field with symbols for both (from which path can be easily found) - having two fields for different levels of processing of essentially the same thing seems somewhat extraneous. This may necessitate exposing something checker-related on the `EmitResolver`.

Reply:
I think making it list of Symbols makes sense. ( though they get converted to paths, I would rather do that work in builder since it is shared with how we handle the imports in the module)